### ISSUE TITLE
Use ssb-blobs-purge, to allow storing all blobs clientside

### ISSUE THREAD

**Issue by Powersource** [arj03/ssb-browser-demo#8]
https://github.com/staltz/ssb-blobs-purge

Would resolve https://github.com/Powersource/ssb-webextension-demo/issues/4

@arj03 do you think this would be a good idea?

**Comment by arj03** [comment]
Yes that would be a great idea! @Powersource

**Comment by KyleMaas** [comment]
How would you figure out what to use for a storage limit?

**Comment by arj03** [comment]
I guess a sane default like maybe 100mb and a way to set this would be a good start I think.

**Comment by KyleMaas** [comment]
It looks like this has to be done at the ssb-browser-core level.  What is the difference between simple-blobs.js and full ssb-blobs?  Because ssb-blobs-purge requires ssb-blobs (or something like it), and simple-blobs.js does not seem to have the APIs it needs to work.

**Comment by arj03** [comment]
Yeah that might be a problem. simple-blobs is a complete replacement of the blobs part of the SSB stack because the existing was littered with fs stuff. So i guess this tasks is a lot more difficuelt because of this sadly :(

**Comment by KyleMaas** [comment]
Hm.  Well, maybe the solution, then, is to try to duplicate (in simple-blobs) the APIs from ssb-blobs which ssb-blobs-purge needs?

**Comment by arj03** [comment]
Yes that would a good way to solve this :-)

**Comment by KyleMaas** [comment]
After attempting this, I'm not sure this is possible.  I got close, but there are two problems I'm running into which I don't see being able to overcome:
* random-access-file doesn't seem to provide a way to see if a file already exists or get the size of a file without creating it.  ssb-blobs-purge uses the size as part of its heuristic.
* As far as I can tell, the timestamp of the blob is not stored.  This is also used by the heuristic.

I've pushed my changes to my ssb-blobs-purge branch, but I think that's about as far as I'm likely to be able to get.  Sorry!

**Comment by arj03** [comment]
Don't be sorry! The missing features in random-access-file might be added later. It's a really good starting point.

For completeness [this](https://github.com/KyleMaas/ssb-browser-core/tree/ssb-blobs-purge) is the branch.

**Comment by KyleMaas** [comment]
So, according to:

https://github.com/random-access-storage/random-access-file/issues/25

...we could potentially check if the blob is zero bytes and use that.  We would still need to figure out the timestamp of the blob.  Is there a good (quick, efficient) way to do that?

**Comment by KyleMaas** [comment]
Getting closer.  I think I've got a workaround for simple-blobs.  However, now I'm blocked on this:

https://github.com/ssbc/ssb-backlinks/issues/22

**Comment by arj03** [comment]
The purge module needs to be converted to db2. Meaning backlinks should be replaced with a seek query. Probably a prefix query (like [key](https://github.com/ssb-ngi-pointer/ssb-db2/blob/25db9fbfa7d8a14614ca9347b742c7529422cfcd/operators/index.js#L20)).

**Comment by KyleMaas** [comment]
https://github.com/staltz/ssb-blobs-purge/issues/1

**Comment by KyleMaas** [comment]
That's live.  Ran into another problem:

https://github.com/staltz/ssb-blobs-purge/issues/3
